pkg/detector/preemption.go

```diff
@@ -283,6 +284,15 @@ func (d *ResourceDetector) HandleDeprioritizedPropagationPolicy(oldPolicy policy
 				return
 			}
 			klog.Infof("Enqueuing PropagationPolicy(%s/%s) in case of PropagationPolicy(%s/%s) priority changes", potentialPolicy.GetNamespace(), potentialPolicy.GetName(), newPolicy.GetNamespace(), newPolicy.GetName())
+			if sortedPotentialKeys == nil {
+				sortedPotentialKeys = rbt.NewWith(int32DescendingComparator)
+			}
+			sortedPotentialKeys.Put(potentialPolicy.ExplicitPriority(), potentialKey)
```

Comment:
I checked the demo in https://github.com/emirpasic/gods/?tab=readme-ov-file#redblacktree:

```go
tree := rbt.NewWithIntComparator() // empty (keys are of type int)

tree.Put(1, "x") // 1->x
tree.Put(2, "b") // 1->x, 2->b (in order)
tree.Put(1, "a") // 1->a, 2->b (in order, replacement)
```

You can see if the key is the same `ExplicitPriority`, it will replace the old value.

So, when you execute `for _, potentialKey := range sortedPotentialKeys.Values()`, some policies has already been overrided

Reply:
Cool finding, thanks.